Website/src/app/list/page.tsx

```diff
@@ -0,0 +1,40 @@
+
+import {cookies} from 'next/headers';
+import {getInitData, getVehicleData} from '@/lib/data';
+import LoginWrapper from "@/components/login_wrap";
+import {InitResponse, Vehicle} from "@/lib/api.website";
+import DynamicList from "@/components/dynlist";
+
+export default async function Home() {
+
+    const token = cookies().get("token")?.value;
+    const track_id = parseInt(cookies().get("track_id")?.value ?? '', 10)
+    const track_selected = !isNaN(track_id);
+    let server_vehicles: Vehicle[];
+    let init_data: InitResponse | undefined;
+    try {
+        init_data = (token && track_selected) ? await getInitData(token, track_id) : undefined;
+        server_vehicles = (token && track_selected) ? await getVehicleData(token, track_id) : [];
+    } catch (e) {
+        console.error('Catched e:', e);
```

Comment:
Error messages should be a bit more informative about the context in which they occurred.